server/src/main/java/org/apache/cassandra/sidecar/coordination/SidecarPeerHealthMonitorTask.java

```diff
@@ -67,19 +65,13 @@ public SidecarPeerHealthMonitorTask(Vertx vertx,
                                         SidecarConfiguration sidecarConfiguration,
                                         SidecarPeerProvider sidecarPeerProvider,
                                         SidecarPeerHealthProvider healthProvider,
-                                        SidecarInstanceCodecs sidecarInstanceCodecs)
+                                        SidecarInstanceCodec sidecarInstanceCodec)
     {
         this.vertx = vertx;
         this.config = sidecarConfiguration.sidecarPeerHealthConfiguration();
         this.sidecarPeerProvider = sidecarPeerProvider;
         this.healthProvider = healthProvider;
-        vertx.eventBus().registerDefaultCodec(SidecarInstance.class, sidecarInstanceCodecs);
-    }
-
-    @NotNull
-    public SidecarPeerHealthProvider.Health status(InstanceMetadata instance)
-    {
-        return status.getOrDefault(instance, SidecarPeerHealthProvider.Health.UNKNOWN);
+        vertx.eventBus().registerDefaultCodec(SidecarInstance.class, sidecarInstanceCodec);
```

Comment:
nit: it is not the best place to register codec for types. For now, it is fine. Maybe comment a todo above this line.